src/GenAPI/Microsoft.DotNet.GenAPI/CSharpFileBuilder.cs

```diff
@@ -174,6 +174,18 @@ private SyntaxNode Visit(SyntaxNode namedTypeNode, INamedTypeSymbol namedType)
                 namedTypeNode = _syntaxGenerator.AddMembers(namedTypeNode, namedType.SynthesizeDummyFields(_symbolFilter));
             }
 
+            // exclude compiler generated members on records
+            if (namedType.IsRecord && namedType.TryGetRecordConstructor(out IMethodSymbol? recordConstructor))
+            {
+                // find the record parameters, so that we can exclude properties
+                var recordParameteterNames = recordConstructor.Parameters.Select(p => p.Name).ToHashSet();
+
+                // exclude compiler generated members, record constructor, and record properties
+                members = members.Where(member => !member.IsCompilerGenerated() &&
+                    !SymbolEqualityComparer.Default.Equals(member, recordConstructor) &&
+                    !recordParameteterNames.Contains(member.Name));
+            }
```

Comment:
Shouldn't that be a filter similar to the ImplicitSymbolFilter: https://github.com/dotnet/sdk/blob/main/src/GenAPI/Microsoft.DotNet.GenAPI/Filtering/ImplicitSymbolFilter.cs?

Reply:
I don't think so - the problem is we can't always reproduce the compiler syntax that will cause it to generate this API.  Good example is `struct record`.  Another example is a source generator that uses this attribute and exposes public API.

Reply:
I think I see what you mean.  I gave a try to refactoring it so that it's efficient enough to do for every member.  Have a look and see if you think that's better.